ext/mbstring/libmbfl/filters/mbfilter_singlebyte.c

```diff
@@ -14,18 +14,25 @@
 
 #include "mbfilter_singlebyte.h"
 
+#if defined(__SSE2__) && defined(__SSSE3__)
```

Comment:
Is SSSE3 part of baseline x64 or does it require a `-march`/`-mcpu` build?

Reply:
Hmm. Thanks for the good question.

I am running GCC 9.3.0. When I run `gcc -dM -E` (`-dM` shows all defines which are automatically set by the preprocessor; `-E` tells GCC to only run the preprocessor and then stop), I can see that `__SSE__` and `__SSE2__` are defined automatically without `-march`, but `__SSSE3__` is only defined if I add `-march=native`.

Hmm. I see that `zend_cpuinfo.h` has some functions for checking CPU support for extended instruction sets at runtime. What would you think of checking the cpuinfo on module load and swapping out a function pointer if the needed instructions are supported?

Reply:
Maybe something to consider for UTF-8, but I don't think it's worthwhile for other encodings.

Reply:
Fair enough. We don't need to merge that commit.